nmoscommon/auth/auth_middleware.py

```diff
@@ -0,0 +1,286 @@
+# Copyright 2019 British Broadcasting Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+import json
+import sys
+import traceback
+
+from time import time
+from flask import abort
+from werkzeug.wrappers import Request, Response
+from werkzeug.exceptions import HTTPException
+from functools import reduce
+from six.moves.urllib.parse import urljoin, parse_qs
+from requests.exceptions import RequestException
+from authlib.jose import jwt, jwk
+from authlib.common.errors import AuthlibBaseError
+from authlib.oauth2.rfc6749.errors import MissingAuthorizationError, UnsupportedTokenTypeError
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+
+from ..mdnsbridge import IppmDNSBridge
+from ..nmoscommonconfig import config as _config
+from .claims_validator import JWTClaimsValidator, generate_claims_options, logger
+
+# DISCOVERY AND CONFIG
+MDNS_SERVICE_TYPE = "nmos-auth"
+OAUTH_MODE = _config.get('oauth_mode', True)
```

Comment:
Whilst this is a change to how things were before, would it be better to default this to 'False' in the same way it is in the nmoscommonconfig file? The same goes for the instance in nmos_auth.py